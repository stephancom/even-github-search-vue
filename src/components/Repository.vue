<template lang="pug">
  b-row.mb-4.px-4.px-md-0
    b-col.col-md-8.col-12.border.p-3
      b-row.pt-3
        b-col
          b-button.float-right(v-if='repo.fork' size='sm' variant='primary') Forked
          a(:href="repo.html_url" target='_blank') {{ repo.full_name }}
      b-row.pt-3
        b-col
          p {{ repo.description }}
    b-col.col-md-2.col-12.border.py-4.text-center
      b-row
        b-col.col-auto.col-md-12 Stars:
        b-col.col-auto.col-md-12.pt-md-2.detail
          span {{ repo.stargazers_count }}
    b-col.col-md-2.col-12.border.py-4.text-center
      b-row
        b-col.col-auto.col-md-12 License:
        b-col.col-auto.col-md-12.pt-md-2.detail
          span(v-if="repo.license") {{ repo.license.name }}
          span(v-else) NONE
</template>

<script>
export default {
  name: "repository",
  props: ["repo"]
};
</script>

<style lang="sass" scoped>
  $even: #377CBE
  a
    color: $even
  .detail
    > a, > span
      font-weight: bold
      color: grey
</style>
