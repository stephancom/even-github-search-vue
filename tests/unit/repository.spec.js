import { shallowMount } from "@vue/test-utils";
import Repository from "@/components/Repository.vue";

describe("Repository.vue", () => {
  const repo = {
    full_name: "boring-coder/some-old-repo",
    html_url: "http://github.com/some_dev/this-repo", // which really should match full_name
    description: "compute all the things",
    stargazers_count: "42",
    license: {
      name: "Arizona Chill License",
    },
    fork: false
  };
  const wrapper = shallowMount(Repository, { propsData: { repo: repo }});
  it("renders repo.full_name when passed", () => {
    expect(wrapper.text()).toMatch("boring-coder/some-old-repo");
  });
  it("renders the html url", () => {
    expect(wrapper.contains('a[href="http://github.com/some_dev/this-repo"]')).toBe(true);
  });
  it("renders the description", () => {
    expect(wrapper.text()).toMatch("compute all the things");
  });
  it("renders the number of stars", () => {
    expect(wrapper.text()).toMatch("42");
  });
  it("renders the license name", () => {
    expect(wrapper.text()).toMatch("Arizona Chill License");
  });
  it("does NOT contain a fork button", () => {
    expect(wrapper.text()).not.toMatch("Forked");
  });
  it("has no buttons", () => {
    expect(wrapper.contains("b-button-stub")).toBe(false);
  });

  describe("forked repo", () => {
    var forkedRepo = repo;
    var forkedWrapper;
    beforeAll(() => {
      forkedRepo.fork = true
      forkedWrapper = shallowMount(Repository, { propsData: { repo: forkedRepo }});
    });
    it("contains forked text", () => {
      expect(forkedWrapper.text()).toMatch("Forked");
    });
    it("contains a button", () => {
      expect(forkedWrapper.contains("b-button-stub")).toBe(true);
    });
    it("contains a forked button", () => {
      expect(forkedWrapper.find("b-button-stub").text()).toMatch("Forked");
    });
  })
})
