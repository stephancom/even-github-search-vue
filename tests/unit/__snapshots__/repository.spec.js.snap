// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository.vue forked repo matches snapshot 1`] = `
<b-row-stub class="mb-4 px-4 px-md-0">
  <b-col-stub class="col-md-8 col-12 border p-3">
    <b-row-stub class="pt-3">
      <b-col-stub>
        <b-button-stub size="sm" variant="primary" class="float-right">Forked</b-button-stub><a href="http://github.com/some_dev/this-repo" target="_blank">boring-coder/some-old-repo</a>
      </b-col-stub>
    </b-row-stub>
    <b-row-stub class="pt-3">
      <b-col-stub>
        <p>compute all the things</p>
      </b-col-stub>
    </b-row-stub>
  </b-col-stub>
  <b-col-stub class="col-md-2 col-12 border py-4 text-center">
    <b-row-stub>
      <b-col-stub class="col-auto col-md-12">Stars:</b-col-stub>
      <b-col-stub class="col-auto col-md-12 pt-md-2 detail"><span>42</span></b-col-stub>
    </b-row-stub>
  </b-col-stub>
  <b-col-stub class="col-md-2 col-12 border py-4 text-center">
    <b-row-stub>
      <b-col-stub class="col-auto col-md-12">License:</b-col-stub>
      <b-col-stub class="col-auto col-md-12 pt-md-2 detail"><span>Arizona Chill License</span></b-col-stub>
    </b-row-stub>
  </b-col-stub>
</b-row-stub>
`;

exports[`Repository.vue matches snapshot 1`] = `
<b-row-stub class="mb-4 px-4 px-md-0">
  <b-col-stub class="col-md-8 col-12 border p-3">
    <b-row-stub class="pt-3">
      <b-col-stub>
        <!----><a href="http://github.com/some_dev/this-repo" target="_blank">boring-coder/some-old-repo</a></b-col-stub>
    </b-row-stub>
    <b-row-stub class="pt-3">
      <b-col-stub>
        <p>compute all the things</p>
      </b-col-stub>
    </b-row-stub>
  </b-col-stub>
  <b-col-stub class="col-md-2 col-12 border py-4 text-center">
    <b-row-stub>
      <b-col-stub class="col-auto col-md-12">Stars:</b-col-stub>
      <b-col-stub class="col-auto col-md-12 pt-md-2 detail"><span>42</span></b-col-stub>
    </b-row-stub>
  </b-col-stub>
  <b-col-stub class="col-md-2 col-12 border py-4 text-center">
    <b-row-stub>
      <b-col-stub class="col-auto col-md-12">License:</b-col-stub>
      <b-col-stub class="col-auto col-md-12 pt-md-2 detail"><span>Arizona Chill License</span></b-col-stub>
    </b-row-stub>
  </b-col-stub>
</b-row-stub>
`;
